integration/send-files/e2e/utils.ts

```diff
@@ -0,0 +1,61 @@
+import { INestApplication } from '@nestjs/common';
+import { IncomingMessage, request, RequestOptions } from 'http';
+
+export const getHttpBaseOptions = async (
+  app: INestApplication,
+): Promise<RequestOptions> => {
+  const url = await app.getUrl();
+  // replace IPv6 localhost with IPv4 localhost alias and split URL on : to get the protocol (http), the host (localhost) and the port (random port)
+  const [protocol, host, port] = url.replace('[::1]', 'localhost').split(':');
```

Comment:
what about

```suggestion
  const { protocol, hostname, port } = new Url(url);
```

![image](https://user-images.githubusercontent.com/13461315/184508746-a8eebcbb-643d-4aa3-a03a-4e138b37c736.png)

and then

```ts
return {
  protocol,
  host: hostname.replace('[::1]', 'localhost'),
  port,
  method: 'GET',
}
```

Reply:
Even better, the `URL` class properly handles `[::1]` so we can just do `new URL(url)` and then `new Url(additionalPath, urlBase)` where needed and pass that directly to `http.request`